assets/js/modules/adsense/util/data-mock.js

```diff
@@ -0,0 +1,221 @@
+	}
+
+	const values = [];
+	const delta = .30;
+
+	for ( const metric of metrics ) {
+		const lastValue = generateMetricValues.memory[ metric ];
+
+		switch ( ADSENSE_METRIC_TYPES[ metric.toUpperCase() ] ) {
+			case METRIC_TALLY: {
+				const newValue = faker.random.number( {
+					min: lastValue ? Math.floor( lastValue * ( 1 - delta ) ) : faker.random.number( { min: 100, max: 900 } ),
+					max: lastValue ? Math.ceil( lastValue * ( 1 + delta ) ) : 1000,
+				} );
+
+				values.push( newValue.toString() );
+				generateMetricValues.memory[ metric ] = newValue;
+				break;
+			}
+			case METRIC_CURRENCY: {
+				const newValue = faker.random.float( {
+					min: lastValue ? Math.floor( lastValue * ( 1 - delta ) ) : faker.random.number( { min: 100, max: 400 } ),
+					max: lastValue ? Math.ceil( lastValue * ( 1 + delta ) ) : 500,
+				} );
+
+				values.push( newValue.toFixed( 2 ) );
+				generateMetricValues.memory[ metric ] = newValue;
+				break;
+			}
+			case METRIC_RATIO: {
+				const newValue = faker.random.float( {
+					min: lastValue ? lastValue * ( 1 - delta ) : faker.random.number( { min: .1, max: .9 } ),
+					max: lastValue ? Math.min( lastValue * ( 1 + delta ), 1 ) : 1,
+				} );
+
+				values.push( newValue.toFixed( 2 ) );
+				generateMetricValues.memory[ metric ] = newValue;
+				break;
+			}
+			default:
+				values.push( '' );
+				break;
+		}
+	}
+
+	return values;
+}
+
+/**
+ * Generates and returns dimension values.
+ *
+ * @since n.e.x.t
+ *
+ * @param {string}         date       The current date.
+ * @param {Array.<string>} dimensions Dimensions list.
+ * @return {Array.<string>} Array of dimension values.
+ */
+function generateDimensionValues( date, dimensions ) {
+	const values = [];
+
+	for ( const dimension of dimensions ) {
+		switch ( dimension.toUpperCase() ) {
+			case 'DATE':
+				values.push( date );
+				break;
+		}
+	}
+
+	return values;
+}
+
+/**
+ * Generates mock data for AdSense reports.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Object} args Report options.
+ * @return {Array.<Object>} An array with generated report.
+ */
+export function getAdSenseMockResponse( args ) {
+	const originalSeedValue = faker.seedValue;
+	const argsHash = parseInt(
+		md5( JSON.stringify( args ) ).substring( 0, 10 ),
+		16,
+	);
+
+	// We set seed for every data mock to make sure that the same arguments get the same report data.
+	// It means that everyone will have the same report data and will see the same widgets in the storybook.
+	// This approach gives us additional flexibility to control randomness on a per widget basis.
+	if ( ! Number.isNaN( argsHash ) ) {
+		faker.seed( argsHash );
+	}
+
+	const metrics = ( Array.isArray( args.metrics ) ? args.metrics : [ args.metrics ] ).filter( ( metric ) => !! metric );
+	const dimensions = ( Array.isArray( args.dimensions ) ? args.dimensions : [ args.dimensions ] ).filter( ( dimension ) => !! dimension );
+
+	const data = {
+		kind: 'adsense#report',
```

Comment:
I believe this is changing with v2 (not much I don't think) so we should probably hold on merging this until #3548 is merged since that one is more important.